### PR TITLE
new script to copy images from docker to...

### PR DESCRIPTION
**Pull request by nickboldt**
new script to copy images from docker to quay.io/eclipse/

Change-Id: I3f99472288daa5b4e62d5701f58d16a72e81d955
Signed-off-by: nickboldt <nboldt@redhat.com>

### CODE REVIEW

File: utils/copyImagesToQuay.txt
Comment (benoitf) [inline utils/copyImagesToQuay.txt:9]: For this image it could be moved by @skabashnyuk

File: utils/copyImagesToQuay.txt
Comment (benoitf) [inline utils/copyImagesToQuay.txt:11]: we can move it from the eclipse Broadway repo

File: utils/copyImagesToQuay.txt
Comment (benoitf) [inline utils/copyImagesToQuay.txt:14]: It's a plugin provided by community it could stay on docker.io

Comment (nickboldt) [reply utils/copyImagesToQuay.txt:14]: except that it'll cause user experience failures so we should copy it over to quay.

File: utils/copyImagesToQuay.txt
Comment (nickboldt) [reply utils/copyImagesToQuay.txt:11]: great, let's do that. Meanwhile...

https://quay.io/repository/eclipse/che--wsskeleton--eclipse-broadway

File: utils/copyImagesToQuay.txt
Comment (nickboldt) [reply utils/copyImagesToQuay.txt:14]: meanwhile https://quay.io/repository/eclipse/che--codercom--code-server

File: utils/copyImagesToQuay.txt
Comment (nickboldt) [reply utils/copyImagesToQuay.txt:9]: meanwhile copied to https://quay.io/repository/eclipse/che--ksmster--che-editor-jupyter

File: utils/copyImagesToQuay.txt
Comment (benoitf) [reply utils/copyImagesToQuay.txt:11]: How could you copy without even approval of this PR ?!!!

File: utils/copyImagesToQuay.txt
Comment (benoitf) [reply utils/copyImagesToQuay.txt:14]: It probably won't cause any pain on these one.but it will cause pain to maintain these useless forks

**Review by benoitf — CHANGES_REQUESTED** [review]
It should not be a blind copy

File: utils/copyImagesToQuay.txt
Comment (nickboldt) [reply utils/copyImagesToQuay.txt:13]: easily. I ran the script locally so that we could get the che release unblocked.

File: utils/copyImagesToQuay.txt
Comment (nickboldt) [reply utils/copyImagesToQuay.txt:16]: 100% agree, but making upstream projects publish to quay instead of docker is something we can tackle in the future.

**Comment by nickboldt** [comment]
> It should not be a blind copy

Why not? it was a blind fetch before so how is this any worse?

File: utils/copyImagesToQuay.txt
Comment (nickboldt) [reply utils/copyImagesToQuay.txt:11]: I see your thumbs-down and raise you a "the release is broken".  :P

File: utils/copyImagesToQuay.txt
Comment (benoitf) [reply utils/copyImagesToQuay.txt:16]: I doubt that release is blocked for plugin's images

**Comment by benoitf** [comment]
> Why not? it was a blind fetch before so how is this any worse?

> I'd appreciate constructive criticism here, or suggestions for a better implementaion, but it feels like all you're saying here is > "this is wrong" without explaining why or how to make it better.

>Please help me be better, don't just tear stuff down.

here are the suggestions, it's not based on `this is wrong`:

1. Some images are maintained on some other registries, then we don't need a blind copy, just change the 'from'
2. Release script might be fixed as well ! For example, is that release script of a given project logs into dockerhub before building images ? Because if so I do not see how it can reach 200 pulls (and we could have multiple login per project as well)
3. You've moved some contributor images, so it means that we'll need to copy again on each new update while for now it's community maintained. Real issue was on the 'core images', the one that you use when deploying che multiple times.
4. if the docker limit was really reached when building plug-in registry or devfile registry, then it does not change anything as these images are using docker.io in their multi-stage builds. But again if the release script would log-in it would not reach this limit as it's like one build per week, so like 20 pulls per project

**Comment by nickboldt** [comment]
> Some images are maintained on some other registries, then we don't need a blind copy, just change the 'from'
Yes, but short term we need a way to reference a valid image from the devfiles/plugins

>  logs into dockerhub before building
Good call, Mykhailo is playing with that for the devfile registry https://github.com/eclipse/che-devfile-registry/commit/51d6ec7ebb831ee5986697c81fadb984ce916e78 and we can expand that out for all the make-release and cico scripts until they can be ported to GH actions

> Real issue was on the 'core images'
Yes, but hitting dockerhub several times to pull digests while building the registries and CSVs means being able to pull ALL the images. My solution was admittedly a stopgap to get us unblocked, but not a permanent fix. 

> docker.io in their multi-stage builds
Yes, and there are app-sre mirrors of alpine images out there we can use but it'll require updating the base images, so it's a non-trivial change to smash through on release day... something for 7.23 in master branch. 

For example: 
* https://github.com/eclipse/che-devfile-registry/blob/master/build/dockerfiles/Dockerfile#L9 (alpine 3.11.5)
* https://quay.io/repository/app-sre/alpine?tab=tags (alpine 3.12.1)

Thanks for the feedback -- it's too bad we didn't tackle the architectural thinking a couple weeks ago before the firefighting started, when we had a bit more time to put something in place before 7.22.0 was due to be released.

File: utils/copyImagesToQuay.txt
Comment (benoitf) [reply utils/copyImagesToQuay.txt:11]: this image should just be dropped from your script

File: utils/copyImagesToQuay.txt
Comment (benoitf) [reply utils/copyImagesToQuay.txt:13]: this image should be removed from the script

File: utils/copyImagesToQuay.txt
Comment (benoitf) [reply utils/copyImagesToQuay.txt:16]: this image should be removed from the script

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
I approve changes if all plug-ins are removed from this script and from quay.io

**Comment by nickboldt** [comment]
```
docker.io/dirigiblelabs/dirigible-openshift:3.4.0
docker.io/ksmster/che-editor-jupyter:5.7.0
docker.io/wsskeleton/eclipse-broadway
docker.io/chinodesuuu/coder:2.1523-vsc1.38.1-che
docker.io/codercom/code-server:3.6.2
docker.io/windup3/rhamt-vscode-extension:java8
```

Deleted:
* https://quay.io/repository/eclipse/che--ksmster--che-editor-jupyter
* https://quay.io/repository/eclipse/che--dirigiblelabs--dirigible-openshift
* https://quay.io/repository/eclipse/che--wsskeleton--eclipse-broadway
* https://quay.io/eclipse/che--codercom--code-server

Never existed as copies:

* docker.io/chinodesuuu/coder:2.1523-vsc1.38.1-che
* docker.io/windup3/rhamt-vscode-extension:java8

**Pull request merged**
